### DISCUSSION TITLE
Unable to access Tokens Table or delete users with associated Token

### DISCUSSION THREAD

**Discussion by LastAeon77** [encode/django-rest-framework#8399] [Potential Issue] [answered]
When I am trying to access Tokens in Admin or when I am trying to delete user, I am hit with this error
![image](https://user-images.githubusercontent.com/56289122/157278552-ee5fcbd6-8963-4f3a-91d7-21ec1232d1f6.png)
I can't delete users no matter what I try, even directly through shell.
![image](https://user-images.githubusercontent.com/56289122/157278874-22da8440-7a91-4a53-a926-79add87a2dc0.png)
I am also unable to specifically access auth_token value through shell 
![image](https://user-images.githubusercontent.com/56289122/157278954-098042c0-f8df-44c5-8bd9-a378b0bf906b.png)
When I try to access the Token table through Admin, this error pops up
![image](https://user-images.githubusercontent.com/56289122/157279387-7091631d-fdbe-4c02-92ad-cb4531256027.png)
When I try to delete users through Admin, This error pops up
![image](https://user-images.githubusercontent.com/56289122/157279545-5b572ce7-1c53-40d8-82f0-be7132b39150.png)
I think it is some logic error in the rest auth token but I am not sure. If anyone could help please do tell!

PS. I do not know if this helps, but as a side note I am using AbstractUser to make my own Users. I am also using simpltjwt for authentications. (I was following this guide to do authentication with another site https://hackernoon.com/110percent-complete-jwt-authentication-with-django-and-react-2020-iejq34ta)

**Comment by hashlash** [comment, marked as answer]
Does your custom `User` model use a `CharField` as its primary key? Could you please provide your custom `User` implementation?

I think it's somewhat similar to: https://stackoverflow.com/questions/31404781/django-and-postgresql-operator-does-not-exist-integer-character-varying.

**Reply by LastAeon77** [reply]
> Does your custom `User` model use a `CharField` as its primary key? Could you please provide your custom `User` implementation?
> 
> I think it's somewhat similar to: https://stackoverflow.com/questions/31404781/django-and-postgresql-operator-does-not-exist-integer-character-varying.

![image](https://user-images.githubusercontent.com/56289122/158068909-8046497d-3c0c-494b-b46b-9b51041b219b.png)
No, all I did with it is add a user favorite color. I have not modified it in any other way. In addition, the error (I presume) comes from the fact that auth_token is a varying char, which is something created from the rest_framework.

**Reply by hashlash** [reply]
`authtoken.user` is created with `OneToOneField` which should use the appropriate column type.

Have you try to recreate the database? Or maybe just re run the `authtoken` and your user module migrations. And make sure there's no new migration changes.

```
$ python manage.py migrate authtoken zero
$ python manage.py migrate <user module> zero
$ python manage.py makemigrations
$ python manage.py migrate
```

**Reply by LastAeon77** [reply]
> `authtoken.user` is created with `OneToOneField` which should use the appropriate column type.
> 
> Have you try to recreate the database? Or maybe just re run the `authtoken` and your user module migrations. And make sure there's no new migration changes.
> 
> ```
> $ python manage.py migrate authtoken zero
> $ python manage.py migrate <user module> zero
> $ python manage.py makemigrations
> $ python manage.py migrate
> ```

I tried this and it still doesn't work. However, you did give me an idea. What I did was I downloaded the database locally, manually deleted the column authtoken_token.user.id, then recreate it into an integer. Afterward, I restored it onto the Heroku database, and everything is fine now. Thank you very much for your patient guidance :D

**Reply by hashlash** [reply]
Nice! Glad to hear that your problem's solved.

You can mark this thread as an answer, so this discussion will be marked as **Answered** and other contributors can focus on other **Unanswered** discussions :D
